homeassistant/components/izone/config_flow.py

```diff
@@ -0,0 +1,34 @@
+"""Config flow for izone."""
+
+import logging
+
+from async_timeout import timeout
+
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_flow
+
+from .constants import DOMAIN
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(hass):
+    from .discovery import async_start_discovery_service
+
+    disco = await async_start_discovery_service(hass)
+
+    try:
+        async with timeout(5):
+            await disco.controller_ready.wait()
+    except TimeoutError:
+        pass
```

Comment:
If it times out, could the disco object be in an invalid/incomplete state, resulting in an exception below? Should we just return 0 here?

Reply:
Nope, if there's a timeout here the disco keeps on running and sends a discovery message later on if a device is discovered. I guess it could be an idea to just fail the config flow and unload the integration though so that it's not just sitting there when it's not needed.

Reply:
I've changed this so if no devices are discovered after 20 seconds, the config flow fails and no further discovery is attempted.